src/node/BrowserFetcher.ts

```diff
@@ -572,23 +573,32 @@ function httpRequest(
 
   const proxyURL = getProxyForUrl(url);
   if (proxyURL) {
-    if (url.startsWith('http:')) {
-      const proxy = URL.parse(proxyURL);
-      options = {
-        path: options.href,
-        host: proxy.hostname,
-        port: proxy.port,
-      };
-    } else {
-      const parsedProxyURL = URL.parse(proxyURL);
-
-      const proxyOptions = {
-        ...parsedProxyURL,
-        secureProxy: parsedProxyURL.protocol === 'https:',
-      } as ProxyAgent.HttpsProxyAgentOptions;
-
-      options.agent = new ProxyAgent(proxyOptions);
-      options.rejectUnauthorized = false;
+    const parsedProxyURL = URL.parse(proxyURL);
+    const proxyOptions = {
+      ...parsedProxyURL,
+      secureProxy: false,
+    };
+    switch (true) {
+      case proxyURL.match(/^socks(4|5|5h)?:/) !== null:
```

Comment:
I firstly added multiple OR (`||`) statemens here but seemed ugly. So I've replaced it with regexp thinking not much performance hit since its only done once at the initial install phase.